src/community/backup-restore/rest/src/main/java/org/geoserver/backuprestore/rest/format/RestoreJSONWriter.java

```diff
@@ -0,0 +1,217 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2016 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.rest.format;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.text.SimpleDateFormat;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.TimeZone;
+import java.util.logging.Level;
+
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.backuprestore.RestoreExecutionAdapter;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.config.util.XStreamPersister;
+import org.geoserver.config.util.XStreamPersister.Callback;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.geoserver.rest.PageInfo;
+import org.geoserver.rest.RestletException;
+import org.restlet.data.Status;
+
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+import net.sf.json.util.JSONBuilder;
+
+/**
+ * Utility class for writing restoreexecutions/statuses/etc... to JSON.
+ * 
+ * Based on Importer {@link ImporterJSONWriter}
+ *  
+ * @author Justin Deoliveira, OpenGeo
+ * @author Alessio Fabiani, GeoSolutions
+ */
+public class RestoreJSONWriter {
```

Comment:
Why not use XStream to handle the in/out? The importer has been openly criticised for not supporting XML, people at DLR could not believe it was not supporting it...